crates/egui/src/viewport.rs

```diff
@@ -0,0 +1,348 @@
+        self.decorations = Some(decorations);
+        self
+    }
+
+    pub fn with_fullscreen(mut self, fullscreen: bool) -> Self {
+        self.fullscreen = Some(fullscreen);
+        self
+    }
+
+    pub fn with_maximized(mut self, maximized: bool) -> Self {
+        self.maximized = Some(maximized);
+        self
+    }
+
+    pub fn with_mimimized(mut self, minimized: bool) -> Self {
+        self.minimized = Some(minimized);
+        self
+    }
+
+    pub fn with_resizable(mut self, resizable: bool) -> Self {
+        self.resizable = Some(resizable);
+        self
+    }
+
+    pub fn with_transparent(mut self, transparent: bool) -> Self {
+        self.transparent = Some(transparent);
+        self
+    }
+
+    /// The icon needs to be wrapped in Arc because will be copied every frame
+    pub fn with_window_icon(mut self, icon: Option<Arc<(u32, u32, Vec<u8>)>>) -> Self {
+        self.icon = Some(icon);
+        self
+    }
+
+    pub fn with_active(mut self, active: bool) -> Self {
+        self.active = Some(active);
+        self
+    }
+
+    pub fn with_visible(mut self, visible: bool) -> Self {
+        self.visible = Some(visible);
+        self
+    }
+
+    pub fn with_title_hidden(mut self, title_hidden: bool) -> Self {
+        self.title_hidden = Some(title_hidden);
+        self
+    }
+
+    pub fn with_titlebar_transparent(mut self, value: bool) -> Self {
+        self.titlebar_transparent = Some(value);
+        self
+    }
+
+    pub fn with_fullsize_content_view(mut self, value: bool) -> Self {
+        self.fullsize_content_view = Some(value);
+        self
+    }
+
+    /// Should be bigger then 0
+    pub fn with_inner_size(mut self, value: Option<Pos2>) -> Self {
+        self.inner_size = Some(value);
+        self
+    }
+
+    /// Should be bigger then 0
+    pub fn with_min_inner_size(mut self, value: Option<Pos2>) -> Self {
+        self.min_inner_size = Some(value);
+        self
+    }
+
+    /// Should be bigger then 0
+    pub fn with_max_inner_size(mut self, value: Option<Pos2>) -> Self {
+        self.max_inner_size = Some(value);
+        self
+    }
+
+    pub fn with_close_button(mut self, value: bool) -> Self {
+        self.close_button = Some(value);
+        self
+    }
+
+    pub fn with_minimize_button(mut self, value: bool) -> Self {
+        self.minimize_button = Some(value);
+        self
+    }
+
+    pub fn with_maximize_button(mut self, value: bool) -> Self {
+        self.maximize_button = Some(value);
+        self
+    }
+
+    /// This currently only work on windows to be disabled!
+    pub fn with_drag_and_drop(mut self, value: bool) -> Self {
+        self.drag_and_drop = Some(value);
+        self
+    }
+
+    pub fn with_position(mut self, value: Option<Pos2>) -> Self {
+        self.position = Some(value);
+        self
+    }
+
+    pub fn with_name(mut self, id: impl Into<String>, instance: impl Into<String>) -> Self {
```

Comment:
All these functions should have docstrings - can we copy the ones from winit maybe?

Reply:
I will copy and adapt winit documentations for every function in ViewportBuilder!